src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -248,6 +250,8 @@ public void openInitialConnection(ConnectionUrl connectionUrl) throws SQLExcepti
     if (this.enableFailoverSetting) {
       initProxy();
     }
+
+    saveSessionState();
```

Comment:
Session state may change after initial connection is open. I think all state transfer logic should be in transferSessionState().

Reply:
If we need to save transaction isolation we need to catch "Connection.setTransactionIsolation" in execute()